src/core/basic_client.rs

```diff
@@ -295,6 +210,14 @@ impl BasicClient {
     ///
     /// If no provider is returned by the service, an client error of `NoProvider`
     /// type is returned.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    ///use parsec_client::BasicClient;
+    ///let mut client = BasicClient::new_naked();
+    ///client.set_default_provider().unwrap();
+    /// ```
```

Comment:
Are these examples actually useful? I mean, you're literally just calling one method without any arguments on a newly created object, I don't think it actually adds anything to the documentation. The ones where the arguments are actually somewhat complex, or that need some setup are probably worth it

Reply:
I think so, it allows users to use the function in their code by just copying and pasting and they are sure it will compile. I also tend to think that even useless documentation/comment is better than nothing.